apps/electron/layers/auth.ts

```diff
@@ -0,0 +1,48 @@
+import ElectronGoogleOAuth2 from '@nosferatu500/electron-google-oauth2';
+import * as http from 'http';
+import fetch from 'node-fetch';
+
+const oauthLoopbackPort = 43102;
+export const electronGoogleOauth = new ElectronGoogleOAuth2(
+  process.env.AFFINE_GOOGLE_CLIENT_ID || '',
+  process.env.AFFINE_GOOGLE_CLIENT_SECRET || '',
+  ['openid', 'https://www.googleapis.com/auth/userinfo.email', 'profile'],
+  {
+    loopbackInterfaceRedirectionPort: oauthLoopbackPort,
+  }
+);
+
+export const startAuthListener = () => {
+  const host = '127.0.0.1';
+  const port = oauthLoopbackPort;
+  const httpHandler: http.RequestListener = (req, res) => {
+    res.writeHead(200, { 'Content-Type': 'text/html' });
+    res.write(`<script>window.location.href = "affine://${req.url}";</script>`);
```

Comment:
start a oauth server to server redirect to affine:// protocal 

will change to affine.pro/callback